libs/poster.py

```diff
@@ -180,6 +180,7 @@ def flush(self):
                 self.packet_count = 0
                 self.event_count = 0
                 self.events = []
+                self.elements.delete_all()
```

Comment:
Is there any way for samples to come in between the time other samples are posted and these elements are cleared? In other words, will we drop samples at all by clearing the elements here?

Reply:
no, we're not dropping samples, once it is flushed, and a new sample/metric comes in, the element will be recreated if not there.

Reply:
Sorry, I meant if there's a race condition. This running in the background vs incoming payloads on a different thread.

Reply:
There is a "with self.lock" protecting this block on line 73 that should address your concern of race condition.